cmake/Modules/LibtorrentMacros.cmake

```diff
@@ -1,80 +1,126 @@
-# _package_dependencies list, which can be used later to generate package config file
-macro(find_public_dependency _name)
-	find_package(${_name} ${ARGN})
-	string(TOUPPER "${_name}" _name_uppercased)
-	if (${_name}_FOUND OR ${_name_uppercased}_FOUND)
-		# Dependencies to be used below for generating Config.cmake file
-		# We don't need the 'REQUIRED' argument there
-		set(_args "${_name}")
-		list(APPEND _args "${ARGN}")
-		list(REMOVE_ITEM _args "REQUIRED")
-		list(REMOVE_ITEM _args "") # just in case
-		string(REPLACE ";" " " _args "${_args}")
-		list(APPEND _package_dependencies "${_args}")
+		if (APPLE)
+			set(CMAKE_REQUIRED_FLAGS "-std=c++11")
+		endif()
+		check_cxx_source_compiles("${ATOMICS_TEST_SOURCE}" HAVE_CXX_ATOMICS_WITHOUT_LIB)
+		check_cxx_source_compiles("${ATOMICS64_TEST_SOURCE}" HAVE_CXX_ATOMICS64_WITHOUT_LIB)
+		if ((NOT HAVE_CXX_ATOMICS_WITHOUT_LIB) OR (NOT HAVE_CXX_ATOMICS64_WITHOUT_LIB))
+			set(CMAKE_REQUIRED_LIBRARIES "atomic")
+			check_cxx_source_compiles("${ATOMICS_TEST_SOURCE}" HAVE_CXX_ATOMICS_WITH_LIB)
+			check_cxx_source_compiles("${ATOMICS64_TEST_SOURCE}" HAVE_CXX_ATOMICS64_WITH_LIB)
+			if ((NOT HAVE_CXX_ATOMICS_WITH_LIB) OR (NOT HAVE_CXX_ATOMICS64_WITH_LIB))
+				message(FATAL_ERROR "No native support for std::atomic, or libatomic not found!")
+			else()
+				message(STATUS "Linking with libatomic for atomics support")
+				unset(CMAKE_REQUIRED_LIBRARIES)
+				target_link_libraries(torrent-rasterbar PUBLIC atomic)
+			endif()
+		endif()
+		if (APPLE)
+			unset(CMAKE_REQUIRED_FLAGS)
+		endif()
 	endif()
+
 endmacro()
 
-# function for parsing version variables that are set in version.hpp file
-# the version identifiers there are defined as follows:
-# #define LIBTORRENT_VERSION_MAJOR 1
-# #define LIBTORRENT_VERSION_MINOR 2
-# #define LIBTORRENT_VERSION_TINY 0
+# This is best effort attempt to propagate whether the library was built with
+# C++11 or not. It affects the ABI of entry. A client building with C++14 and
+# linking against a libtorrent binary built with C++11 can still define
+# TORRENT_CXX11_ABI
+macro(libtorrent_cxx11_abi_issue_workaround)
 
-function(read_version _verFile _outVarMajor _outVarMinor _outVarTiny)
-	file(STRINGS ${_verFile} verFileContents REGEX ".+LIBTORRENT_VERSION_[A-Z]+.[0-9]+.*")
-# 	message(STATUS "version file contents: ${verFileContents}")
-	# the verFileContents variable contains something like the following:
-	# #define LIBTORRENT_VERSION_MAJOR 1;#define LIBTORRENT_VERSION_MINOR 2;#define LIBTORRENT_VERSION_TINY 0
-	set(_regex ".+_MAJOR +([0-9]+);.+_MINOR +([0-9]+);.+_TINY +([0-9]+)")
-	 # note quotes around _regex, they are needed because the variable contains semicolons
-	string(REGEX MATCH "${_regex}" _tmp "${verFileContents}")
-	if (NOT _tmp)
-		message(FATAL_ERROR "Could not detect project version number from ${_verFile}")
+	if ("${CMAKE_CXX_STANDARD}" STREQUAL "11")
+		target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_CXX11_ABI)
```

Comment:
this would make it into the pkg-config file too, is that right?

Reply:
@arvidn Yes, just confirmed it (previously, it did as well).